packages/utils/src/mergeObjects.ts

```diff
@@ -5,21 +5,31 @@ import { GenericObjectType } from "./types";
  *
  * @param obj1 - The first object to merge
  * @param obj2 - The second object to merge
- * @param [concatArrays=false] - Optional flag that, when true, will cause arrays to be concatenated
+ * @param [concatArrays=false] - Optional flag that, when true, will cause arrays to be concatenated. Use
+ *          "preventDuplicates" to merge arrays in a manner that prevents any duplicate entries from being merged.
  * @returns - A new object that is the merge of the two given objects
  */
 export default function mergeObjects(
   obj1: GenericObjectType,
   obj2: GenericObjectType,
-  concatArrays = false
+  concatArrays: boolean | "preventDuplicates" = false
 ) {
   return Object.keys(obj2).reduce((acc, key) => {
     const left = obj1 ? obj1[key] : {},
       right = obj2[key];
     if (obj1 && key in obj1 && isObject(right)) {
       acc[key] = mergeObjects(left, right, concatArrays);
     } else if (concatArrays && Array.isArray(left) && Array.isArray(right)) {
-      acc[key] = left.concat(right);
+      let toMerge = right;
+      if (concatArrays === "preventDuplicates") {
+        toMerge = right.reduce((result, value) => {
+          if (!left.includes(value)) {
```

Comment:
This should work just fine for `ErrorSchema`s because it's an Object of objects or string arrays, but I'm concerned someone may try to use this and expect this to deduplicate with deep equality. What do you think? Either way, please update the @param doc for "preventDuplicates" to indicate the behavior.

Reply:
done